shop.lua

```diff
@@ -5,6 +5,30 @@
 
 local exchange_shop = {}
 
+-- Tool wear aware replacement for contains_item.
+local function list_contains_item(inv, list, stack)
+	for i, list_stack in pairs(inv:get_list(list)) do
+		if list_stack:get_name()  == stack:get_name()  and
+		   list_stack:get_count() >= stack:get_count() and
+		   list_stack:get_wear()  <= stack:get_wear() then
+			return true, i
```

Comment:
These are both good catches. I've made the requested changes. ATM they are in a separate commit for clarity but I will squash them into one upon request.